test/vol.c

```diff
@@ -1176,6 +1538,510 @@ test_basic_datatype_operation(void)
+
+    /* Try finding a non-existent optional VOL subclass operation */
+    H5E_BEGIN_TRY
+    {
+        ret = H5VLfind_opt_operation(H5VL_SUBCLS_DATASET, "fail", &op_val);
+    }
+    H5E_END_TRY;
+    if (FAIL != ret)
+        FAIL_PUTS_ERROR("should not be able to find a non-existent optional operation");
+
+    /* Try unregistering a non-existent optional VOL subclass operation */
+    H5E_BEGIN_TRY
+    {
+        ret = H5VLunregister_opt_operation(H5VL_SUBCLS_DATASET, "fail");
+    }
+    H5E_END_TRY;
+    if (FAIL != ret)
+        FAIL_PUTS_ERROR("should not be able to unregister a non-existent optional operation");
+
+    /* Register & test calling optional operations for each valid VOL subclass */
+    /* (Table-driven, with test_params array) */
+    for (u = 0; u < NELMTS(test_params); u++)
+        /* Exercise appropriate callback, for each VOL subclass */
+        if (exercise_reg_opt_oper(fake_vol_id, reg_opt_vol_id, test_params[u].subcls,
+                                  test_params[u].subcls_name, test_params[u].id_type,
+                                  test_params[u].reg_opt_op) < 0)
+            TEST_ERROR;
+
+    /* Unregister the VOL connectors */
+    if (H5VLunregister_connector(fake_vol_id) < 0)
+        TEST_ERROR;
+    if (H5VLunregister_connector(reg_opt_vol_id) < 0)
+        TEST_ERROR;
+
+    PASSED();
+
+    return SUCCEED;
+
+error:
+    H5E_BEGIN_TRY
+    {
+        H5VLunregister_connector(fake_vol_id);
+        H5VLunregister_connector(reg_opt_vol_id);
+    }
+    H5E_END_TRY;
+
+    return FAIL;
+} /* end test_register_opt_operation() */
+
+/*-------------------------------------------------------------------------
+ * Function:    test_async_vol_props()
+ *
+ * Purpose:     Test properties related to asynchronous VOL connector operation
+ *
+ * Note:        Overrides the HDF5_VOL_CONNECTOR environment variable, to
+ *              provide stable testing environment.
+ *
+ * Return:      SUCCEED/FAIL
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+test_async_vol_props(void)
+{
+    hid_t                    fapl_id = H5I_INVALID_HID;
+    hid_t                    vol_id  = H5I_INVALID_HID;
+    H5VL_pass_through_info_t passthru_info;
+    unsigned                 cap_flags    = 0;
+    char *                   conn_env_str = NULL;
+
+    TESTING("Async VOL props");
+
+    /* Retrieve the file access property for testing */
+    fapl_id = h5_fileaccess();
+
+    /* Test 'capability flags' property */
+
+    /* Test query w/NULL for cap_flags parameter */
+    if (H5Pget_vol_cap_flags(fapl_id, NULL) < 0)
+        FAIL_STACK_ERROR;
+
+    /* Override possible environment variable & re-initialize default VOL connector */
+    conn_env_str = HDgetenv("HDF5_VOL_CONNECTOR");
+    if (conn_env_str) {
+        if (NULL == (conn_env_str = HDstrdup(conn_env_str)))
+            TEST_ERROR
+        if (HDunsetenv("HDF5_VOL_CONNECTOR") < 0)
```

Comment:
Seems Windows is unhappy about this, since setenv/unsetenv aren't standard C. @derobins Is there a Windows equivalent for these? They're POSIX 2001, so I figure there probably are.

Reply:
@jhendersonHDF - I believe that I've fixed this, can you please re-test?

Reply:
I was just paying attention to the CI checks. Since they're running again we should see soon.